sqlparse/joins.py

```diff
@@ -0,0 +1,60 @@
+import itertools
+
+
+# this data structure is a kind of grammar for allowed SQL joins. it expands
+# to a sequence of keyword tuples representing valid joins as follows:
+# - tuples expand to a union of the types encoded in each element
+#   - empty string element means the "group" is optional
+# - lists represent product of the types encoded in each element
+# - strings represent a literal option for an element of the type
+#
+# it is mainly modeled after the duckdb join syntax documented here:
+#   https://duckdb.org/docs/sql/query_syntax/from#syntax
+JOIN_TYPES = (
+    'STRAIGHT_JOIN',
+    [
+        (
+            'CROSS',
+            'POSITIONAL',
+            [
+                ('', 'NATURAL', 'ASOF'),
+                ('', 'INNER', [('LEFT', 'RIGHT', 'FULL'), ('', 'OUTER')]),
+            ],
+        ),
+        'JOIN',
+    ],
+)
+
+
+# generates all valid join types as a sequence of tuples of keywords
+def enumerate_types(join_types=JOIN_TYPES):
```

Comment:
Is it worth it to have a test that validates what this generates? Could see it being nice to have it visually represented as a sanity check